packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -716,18 +718,28 @@ export class StripeHelper {
       };
 
       try {
-        const invoice = await this.previewInvoice({
-          country,
-          priceId,
-          promotionCode,
-        });
+        const { currency, discount, total, total_discount_amounts } =
+          await this.previewInvoice({
+            country,
+            priceId,
+            promotionCode,
+          });
+
+        if (discount && total_discount_amounts) {
+          couponDetails.discountAmount = total_discount_amounts[0].amount;
+        }
 
-        if (invoice?.discount && invoice?.total_discount_amounts) {
-          couponDetails.discountAmount =
-            invoice.total_discount_amounts[0].amount;
+        const minAmount = getMinimumAmount(currency);
+        if (total !== 0 && minAmount && total < minAmount) {
+          throw error.invalidPromoCode(promotionCode);
+        }
+      } catch (error) {
+        if (
+          error instanceof AppError &&
+          error.errno === AppError.ERRNO.INVALID_PROMOTION_CODE
+        ) {
+          throw error;
```

Comment:
Since we are only re-throwing the invalid promotion code error here does that mean if `previewInvoice` or `getMinimumAmount` throws an error, we simply won't know about it?

Reply:
A good catch. Added an else, which will report the error to Sentry.